wordpress-mobile/WordPress-iOS
# Migrate fetching of Page Layouts call from WPiOS to WPKit

### Description
This is a simple refactoring to move the api calls `/wpcom/v2/sites/\(blogID)/block-layouts` and `/wpcom/v2/common-block-layouts` out of WPiOS and into WordPressKit. 

**Related PR:**
`WordPress-Kit:` https://github.com/wordpress-mobile/WordPressKit-iOS/pull/294

### Testing Details
📓 This is easiest to test with a fresh install so you know you aren't just seeing cached data in the app. 

<details>
<summary><strong>To disable or enable the development version of Modal Layout Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <img width=300 src="https://user-images.githubusercontent.com/3384451/90816133-c8b10580-e2f9-11ea-8584-6bbfd7e523e2.gif" />
</details>

When enabled, the Layout Picker should show when creating a new page from My Site and from the Page list.

- Navigating to the Modal Layout Picker
- Expect page layouts to be fetched and to load.

### PR submission checklist:
